internal/controller/grafanaorganization_controller.go

```diff
@@ -129,22 +134,60 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 	}
 
 	// Update CR status if anything was changed
-	if organization.ID != grafanaOrganization.Status.OrgID {
-		grafanaOrganization.Status.OrgID = organization.ID
+	if err = r.updateOrgStatus(ctx, grafanaOrganization, organization.ID); err != nil {
```

Comment:
Why is update OrgStatus creating the datasources? Also, I think we need to update once the organization exists or we might recreate it?

Reply:
> Why is update OrgStatus creating the datasources? 

Well, in order for the or CRs' status to be updated with the list of datasources, we first need to create those datasources objects before updating. 
If you prefer I can outsource the code for the datasource creation and/or the following status update in another function.

>  I think we need to update once the organization exists or we might recreate it?

Isn't that already the case ?

Reply:
So, here you update with the datasource info right? 

But if this fails to create the datasources, you will have 1 organization wihtout a status while it is created on grafana so any following reconciliation will fail due to a same name error right?

What I think is that we should:

CRUD organization

- > Update status

CRUD datasources
- > update status again

Configure org-mapping and restart grafana

Reply:
> But if this fails to create the datasources, you will have 1 organization wihtout a status while it is created on grafana so any following reconciliation will fail due to a same name error right?

Indeed :/